### PR TITLE
Implementation of project features

### PR DESCRIPTION
**Pull request by YuvrajM108**
In this pull request, I did the following:

- Selected an API to fetch data from.
- Created components for the application and added routes for navigating to the components.
- Created a store for the application.
- Created actions and action creators for fetching data from the API.
- Created reducers for each component and combined the reducers in the store.
- Connected actions and reducers to the components and outputted data from the store to the components.
- Styled the components according to the given design.
- Created unit and integration tests.
- Deployed the project on Netlify.
- Made a video presentation of the project ([click here to view the presentation](https://www.loom.com/share/f1a0cf88f14f4c78920da483d8a20a3a))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,64 @@
...
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
-
-### Deployment
-
-This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
-
-### `yarn build` fails to minify
-
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+This project is [MIT](./MIT.md) licensed.
```
Comment (henatan99) [inline README.md:68]: - [x] Kindly add an "Acknowledgements" section or similar that gives appropriate credit to Nelson Sakwa the author of the original design, as required by the Creative Commons License. :pray:

File: src/__mocks__/mockAPIRequest.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  axios.get('https://api.carbonintensity.org.uk/regional/regionid/16')
+    .then((res) => res.data)
+    .catch((error) => error);
+};
+
+export const getWalesDetails = () => {
+  axios.get('https://api.carbonintensity.org.uk/regional/regionid/17')
+    .then((res) => res.data)
+    .catch((error) => error);
+};
```
Comment (henatan99) [inline src/__mocks__/mockAPIRequest.js:25]: - [ ] OPTIONAL: I think you could have made just a single request with the URL `https://api.carbonintensity.org.uk/regional/regionid/` and filter with region_id for a particular region. Your code could be way lighter and cleaner. :pray:

File: src/App.css [hunk 3]
```diff
@@ -1,38 +1,84 @@
...
+.nav-icon {
+  font-size: 28px;
+  margin: 0 5px 0 5px;
+  color: white;
 }
 
-.App-header {
-  background-color: #282c34;
-  min-height: 100vh;
+a.nav-link {
```
Comment (henatan99) [inline src/App.css:27]: - [x] While you are free to pick a base color of your own it is a requirement to maintain layout: composition and space between elements of the original design. Kindly compare the layout for this element in the original design and make changes to your design to match it. 

image
image

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @YuvrajM108 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [x] I could not find a video link either in your PR summary or in the readme. The project requires you once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:
  - a description of the project.
  - a demo of the project features.
  - you should also highlight some interesting piece of code or something you built that you are very proud of.


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**